### PR TITLE
[#166334937] Add username column to users table

### PR DESCRIPTION
**Pull request by mogds**
What
------

This adds a username column to the users table and updates the API surface/validation to match.

Not all users have an email address, such as the admin user, but we want to maintain a display name for all users. We add the username column for this purpose.

In the future, we will need to make the username field non-nullable, but only once other work to insert non-email-addressed users in to paas-accounts at different times.

How to review
-------

- Code Review
- cf push to your dev env and test http endpoints

Who can review
------

Not me or @AP-Hunt

### CODE REVIEW

**Comment by AP-Hunt** [comment]
> I'd like it if there was some sort of test where the `email` and `username` differed, to guard against a future mistake where the two fields got confused, and since that's part of how we plan to use the API. Otherwise the code looks good to me 🎉

We have a deliberate test of that [here](https://github.com/alphagov/paas-accounts/pull/10/files#diff-979c7fe90df5dfe3d23900920ec84169R73) but I agree that it should probably be a test case of its own.

**Comment by 46bit** [comment]
Ah sorry, I missed that one.

File: api/patch_user_handler.go [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
 
-			err = db.PatchUser(user)
-			if err != nil {
-				return err
-			}
+			return err
+		}
+
+		if user.Username == nil {
+			user.Username = &payload.Username
```
Comment (46bit) [inline api/patch_user_handler.go]: This silently fails when `PATCH`ing the username of a user with a username already set. I'd prefer to return an error instead, but open to discussion :)

Comment (AP-Hunt) [reply api/patch_user_handler.go]: This was deliberate. We figure we don't really ever want usernames to be updated (we think), but we also have a scenario where we have existing users in the database that don't have a username set. Setting the username, but only if it hasn't been set yet, allows us to accommodate that. It's useful when migrating users, and when rolling out the change to new environments.

We'll be introducing a change to paas-accounts deployment for the latter.

Comment (46bit) [reply api/patch_user_handler.go]: I decided this was very much personal preference and not a big deal. :)

**Pull request merged**